constraint-solver/src/grouped_expression.rs

```diff
@@ -893,7 +893,7 @@ fn combine_range_constraints<T: RuntimeConstant, V: Ord + Clone + Eq + Hash + Di
             // constraint, and thus we want it to only hit the "single value" case.
             let complete = rc1.try_to_single_value().is_some()
                 && rc2.try_to_single_value().is_some()
-                && rc.range_width() <= 2.into();
+                && rc.size() <= 2.into();
```

Comment:
Would it be equivalent to check if `rc1 == rc2`? That's the only case I can think of where none of the two are concrete values but the disjunction only has two elements.

Reply:
The problem is when we have two concrete values but they are not next to each other. Then the current range constraint system cannot always fully capture it (it might using bit masks).